From 565efa61631ef01b5c5c29c0132aa7bcb327b331 Mon Sep 17 00:00:00 2001
From: Adrien Plazas <kekun.plazas@laposte.net>
Date: Tue, 9 Nov 2021 10:47:06 +0100
Subject: [PATCH] utils: Add gcal_get_abbreviated_month_name()

It will be used to help the month names fit phone screens.
---
 src/utils/gcal-utils.c | 31 +++++++++++++++++++++++++++++++
 src/utils/gcal-utils.h |  2 ++
 2 files changed, 33 insertions(+)

diff --git a/src/utils/gcal-utils.c b/src/utils/gcal-utils.c
index e57c064d..0de5e94f 100644
--- a/src/utils/gcal-utils.c
+++ b/src/utils/gcal-utils.c
@@ -101,6 +101,23 @@ month_item[12] =
 #endif
 };
 
+static const gint
+abbreviated_month_item[12] =
+{
+  _NL_ABALTMON_1,
+  _NL_ABALTMON_2,
+  _NL_ABALTMON_3,
+  _NL_ABALTMON_4,
+  _NL_ABALTMON_5,
+  _NL_ABALTMON_6,
+  _NL_ABALTMON_7,
+  _NL_ABALTMON_8,
+  _NL_ABALTMON_9,
+  _NL_ABALTMON_10,
+  _NL_ABALTMON_11,
+  _NL_ABALTMON_12
+};
+
 #define SCROLL_HARDNESS 10.0
 
 /**
@@ -131,6 +148,20 @@ gcal_get_month_name (gint i)
   return nl_langinfo (month_item[i]);
 }
 
+/**
+ * gcal_get_abbreviated_month_name:
+ * @i: the month index
+ *
+ * Retrieves the abbreviated month name.
+ *
+ * Returns: (transfer full): the abbreviated month name
+ */
+gchar*
+gcal_get_abbreviated_month_name (gint i)
+{
+  return nl_langinfo (abbreviated_month_item[i]);
+}
+
 /**
  * gcal_get_surface_from_color:
  * @color: a #GdkRGBA
diff --git a/src/utils/gcal-utils.h b/src/utils/gcal-utils.h
index d2d3631d..28b87694 100644
--- a/src/utils/gcal-utils.h
+++ b/src/utils/gcal-utils.h
@@ -49,6 +49,8 @@ gchar*               gcal_get_weekday                            (gint
 
 gchar*               gcal_get_month_name                         (gint                i);
 
+gchar*               gcal_get_abbreviated_month_name             (gint                i);
+
 cairo_surface_t*     gcal_get_surface_from_color                 (const GdkRGBA      *color,
                                                                   gint                size);
 
-- 
2.30.2

